builder/vmware/common/ssh.go

```diff
@@ -4,30 +4,58 @@ import (
 	"errors"
 	"fmt"
 	"log"
+	"net"
 
 	"github.com/hashicorp/packer/helper/multistep"
 )
 
 func CommHost(config *SSHConfig) func(multistep.StateBag) (string, error) {
 	return func(state multistep.StateBag) (string, error) {
 		driver := state.Get("driver").(Driver)
+		comm := config.Comm
 
-		if config.Comm.SSHHost != "" {
-			return config.Comm.SSHHost, nil
+		// Figure out which protocol we're using to communicate with the
+		// guest, and determine the correct port to handshake with.
+		port := comm.SSHPort
```

Comment:
The communicator config itself has a helper function to retrieve the correct host/port based on communicator type: https://github.com/hashicorp/packer/blob/master/helper/communicator/config.go#L318-L340 

use  `port := comm.Port()` and you can bypass this set of conditionals. We also have a comm.Host() that does the same thing for retrieving SSHHost vs WinRMHost, which is probably safer than using SSHHost in line 20.

Reply:
Ah awesome. Good to know.. Fixing it rn.

Reply:
Yeah. I copied that logic from driver_esx5.go. I'll update that relevant lines in that file too.

Reply:
rad, thanks!

Reply:
I'll do the final review for this tomorrow, so don't rush to get the commit in right now.